blockdevice.go

```diff
@@ -157,17 +182,26 @@ func blockdevice_2_0(source map[string]interface{}) (*blockdevice, error) {
 	// From here we know that the map returned from the schema coercion
 	// contains fields of the right type.
 
+	var filesystem *filesystem
+	if fsSource := valid["filesystem"]; fsSource != nil {
```

Comment:
This would be better:

if fsSource, ok := valid["filesystem"].(map[string]interface{}); ok {
	filesystem, err = filesystem2_0(fsSource)